python/lsst/sims/maf/metrics/optimalM5Metric.py

```diff
@@ -0,0 +1,65 @@
+from .baseMetric import BaseMetric
+from .simpleMetrics import Coaddm5Metric
+import numpy as np
+
+__all__ = ['OptimalM5Metric']
+
+
+class OptimalM5Metric(BaseMetric):
+    """
+    Compare the co-added depth of the survey to one where
+    all the observations were taken on the meridian.
+    """
+
+    def __init__(self, m5Col='fiveSigmaDepth', optM5Col='m5Optimal',
+                 normalize=False, **kwargs):
+        """
+        Parameters
+        ----------
+        m5Col : str ('fiveSigmaDepth')
+            Column name that contains the five-sigma limiting depth of
+            each observation
+        optM5Col : str ('m5Optimal')
+            The column name of the five-sigma-limiting depth if the
+            observation had been taken on the meridian.
+        normalize : bool (False)
+            If False, metric returns how many more observations would need
+            to be taken to reach the optimal depth.  If True, the number
+            is normalized by the total number of observations already taken
+            at that position.
+
+        Returns
+        --------
+        out : float
+            If normalize is True, output is the percentage of time the survey
+            is behind an optimal meridian-scanning survey. So if a 10-year
+            survey is at 20%, it would need to run for 12 years to reach the
+            same depth as a 10-year meridian survey.
+            If normalize is False (default), `out` is just the number of
+            additional observations the survey needs to catch up to optimal.
```

Comment:
These comments (parameters and returns) actually go directly on the class comments, because __init__ comments don't actually get propagated to the API docs. 
Returns should not give the name "out" -- just the type. i.e.
Returns
----------
float
    (also -- I don't understand the comment return value ... unless this is a metric which is intended to be run with the unislicer? if so, that would be good to add to the comments). 
Does this return the number of additional observations (or the percentage of additional observations) required to make the depth match a 10-year meridian survey?  -- and it could be at a particular point in the sky  (healpix) OR over the whole sky (unislicer)? I think I was getting confused by some of the language here which seems to refer to the *entire* survey, not just a part.

(also .. 10 year survey at 20% -> need to run for 12 years to reach the same depth as 10 year meridian survey seems odd .. if it was at 20% wouldn't it need 5x more observations to reach 100%, so would need 50 years? what am i missing?)  (similar example would be nice for the normalize is False option though).